src/huggingface_hub/hf_api.py

```diff
@@ -581,7 +580,7 @@ def whoami(self, token: Optional[str] = None) -> Dict:
         path = f"{self.endpoint}/api/whoami-v2"
         r = requests.get(path, headers={"authorization": f"Bearer {token}"})
         try:
-            _raise_with_request_id(r)
+            _raise_for_status(r)
```

Comment:
It does not make sense to raise `RepoNotFound` error on the `whoami` endpoint, we should stick to `_raise_with_request_id` or similar 😅 

`_raise_for_status` should only be used on repo endpoints, otherwise confusing errors can be raised

Reply:
`_raise_for_status` would not raise a `RepoNotFound`. That is just raised if that's in the error code returned in the headers. Since that's not set, it gives a usual error

```
requests.exceptions.HTTPError: Invalid user token. If you didn't pass a user token, make sure you are properly logged in by executing `huggingface-cli login`, and if you did pass a user token, double-check it's correct.
```

Also note that below the error is raised as `HttpError` with custom message

Reply:
`_raise_for_status` still raises `RepoNotFound` on a 401 (not authenticated) error, no?
https://github.com/huggingface/huggingface_hub/pull/924/files#diff-767725f8f18383b90d2c454bbdcf39cdf090122933b71c3eda71268f1fb2922fR96-R103

This is correct on repo endpoints, but not for other endpoints (`whoami`, but also the pre-signed URLs for LFS uploads)

Am I missing something? 😦

Reply:
Ah yes, that's right. For `whoami` the error is still handled and a `HTTPError` is handled though, but I guess the error stack is confusing. You're right!

https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/hf_api.py#L583-L591

Reply:
Ah, thanks for the pointer!

Ok, then I guess it's not as big as a deal as I thought it was 😅

Reply:
Still I think the error stack is confusing, so I'll revert those changes 

```
 python test.py 
('401 Client Error: Repository Not Found for url: https://huggingface.co/api/whoami-v2. If the repo is private, make sure you are authenticated. (Request ID: ZcxhTMDv6NxQlsUcZYgql)',)
Traceback (most recent call last):
  File "/home/osanseviero/miniconda3/envs/huggingface_hub/lib/python3.9/site-packages/huggingface_hub/utils/_errors.py", line 69, in _raise_for_status
    response.raise_for_status()
  File "/home/osanseviero/miniconda3/envs/huggingface_hub/lib/python3.9/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://huggingface.co/api/whoami-v2

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/osanseviero/miniconda3/envs/huggingface_hub/lib/python3.9/site-packages/huggingface_hub/hf_api.py", line 583, in whoami
    _raise_for_status(r)
  File "/home/osanseviero/miniconda3/envs/huggingface_hub/lib/python3.9/site-packages/huggingface_hub/utils/_errors.py", line 103, in _raise_for_status
    raise e
huggingface_hub.utils._errors.RepositoryNotFoundError: 401 Client Error: Repository Not Found for url: https://huggingface.co/api/whoami-v2. If the repo is private, make sure you are authenticated. (Request ID: ZcxhTMDv6NxQlsUcZYgql)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/osanseviero/Desktop/workspace/huggingface_hub/test.py", line 5, in <module>
    HfApi().whoami("hf_test")
  File "/home/osanseviero/miniconda3/envs/huggingface_hub/lib/python3.9/site-packages/huggingface_hub/hf_api.py", line 585, in whoami
    raise HTTPError(
requests.exceptions.HTTPError: Invalid user token. If you didn't pass a user token, make sure you are properly logged in by executing `huggingface-cli login`, and if you did pass a user token, double-check it's correct.
```